File: Week 07/id_318/LeetCode_190_318.js
/*
 * @lc app=leetcode id=190 lang=javascript
 *
 * [190] Reverse Bits
 */

// @lc code=start
/**
 * @param {number} n - a positive integer
 * @return {number} - a positive integer
 */
var reverseBits = function(n) {
    let reverse_n = 0;
    for (let i = 0; i < 32; ++i) {
        reverse_n = (reverse_n << 1) + (n & 1);
        n >>= 1;
    }
    return reverse_n >>> 0;
};
// @lc code=end

File: Week 07/id_318/LeetCode_190_318.cpp
/*
 * @lc app=leetcode id=190 lang=cpp
 *
 * [190] Reverse Bits
 */

// @lc code=start
class Solution {
public:
    uint32_t reverseBits(uint32_t n) {
        int i = 32;
        uint32_t reverse_n = 0;
        while(i--) {
            reverse_n = (reverse_n << 1) + (n & 1);
            n >>= 1;
        }
        return reverse_n;
    }
};
// @lc code=end

